CMakeLists.txt

```diff
@@ -36,6 +36,25 @@ if(CapnProto_VERSION STREQUAL "0.7.0"
   ")
 endif()
 
+# Check for Cap'n Proto / Clang / C++20 incompatibility
+# Cap'n Proto 0.9.x and 0.10.x are incompatible with Clang 16+ when using C++20
+# due to P2468R2 implementation. This was fixed in Cap'n Proto 1.0+.
+# See: https://github.com/bitcoin-core/libmultiprocess/issues/199
+if((CapnProto_VERSION VERSION_GREATER_EQUAL "0.9.0") AND
+   (CapnProto_VERSION VERSION_LESS "1.0.0") AND
+   (CMAKE_CXX_COMPILER_ID STREQUAL "Clang") AND
+   (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "16") AND
+   (CMAKE_CXX_STANDARD EQUAL 20))
+  message(FATAL_ERROR
+    "Cap'n Proto ${CapnProto_VERSION} is incompatible with Clang ${CMAKE_CXX_COMPILER_VERSION} when using C++20.\n"
+    "To resolve this issue, choose one of the following:\n"
+    "  1. Upgrade to Cap'n Proto version 1.0 or newer (recommended)\n"
+    "  2. Downgrade to Cap'n Proto version 0.8 or earlier\n"
```

Comment:
> Downgrade to Cap'n Proto version 0.8 or earlier

I think this should say `0.8.1`, so it's clear we aren't suggesting to downgrade to 0.8.0, which wont work, given the check for CVEs above. Although I think ideally, we wouldn't suggest downgrading at all.